app/frontend/components/pages/delegations/delegatePage.tsx

```diff
@@ -54,48 +97,100 @@ const Delegate = ({withAccordion, title}: Props): h.JSX.Element => {
     calculatingDelegationFee: state.calculatingDelegationFee,
     delegationFee: state.shelleyDelegation.delegationFee,
-    delegationValidationError: state.delegationValidationError,
     txSuccessTab: state.txSuccessTab,
     gettingPoolInfo: state.gettingPoolInfo,
     isShelleyCompatible: state.isShelleyCompatible,
     poolRecommendation: getSourceAccountInfo(state).poolRecommendation,
-    pool: getSourceAccountInfo(state).shelleyAccountInfo.delegation,
     isBigDelegator: isBigDelegatorSelector(state),
+    validStakepoolDataProvider: state.validStakepoolDataProvider,
   }))
-  const {confirmTransaction} = useActions(actions)
+  const {confirmTransaction, updateStakePoolIdentifier} = useActions(actions)
+
+  // TODO: this reference always changes inside "useEffect" which causes infite loop.
+  // Till this is investigated, this workaround is used instead of "eslint-disable-..."
+  const _updateStakePoolIdentifier = useRef(updateStakePoolIdentifier)
+
+  const [fieldValue, setFieldValue] = useState('')
+  const [error, setError] = useState<Error>(null)
+
+  const handleOnInput = (event: any): void => {
+    const fieldValue: string = event?.target?.value
+    setFieldValue(fieldValue)
+
+    // Note: this could be debounced
+    if (fieldValue) {
+      const {poolHash, error} = validateInput(fieldValue, validStakepoolDataProvider)
+      if (!error) {
+        /*
+        Relates to REFACTOR (calculateFee):
+        This component should store all `txPlan` and `error` states & render confirmation modal
+        when user clicks on "Delegate". This could greatly simplify global state flow. For now
+        using hybrid solution involving `updateStakePoolIdentifier` (should be later removed)
+        */
+        _updateStakePoolIdentifier.current(poolHash)
+        setError(null)
+      } else {
+        setError(error)
+      }
+    }
+  }
+
+  // init "stake pool input" and refresh it when "currentDelegation" changes
+  useEffect(() => {
+    const recommendedPoolHash =
+      poolRecommendation?.recommendedPoolHash || currentDelegation?.poolHash
 
-  /*
-  REFACTOR (calculateFee):
-  Calculation of "fee" should be fired from there explicitely using "useEffect"
-  when component mounts.
-  */
+    if (recommendedPoolHash) {
+      const {poolHash, error} = validateInput(recommendedPoolHash, validStakepoolDataProvider)
+      if (error) {
+        setError(error)
+      } else {
+        _updateStakePoolIdentifier.current(poolHash)
+      }
+      setFieldValue(recommendedPoolHash)
+    }
+  }, [
+    currentDelegation,
+    _updateStakePoolIdentifier,
+    poolRecommendation,
+    validStakepoolDataProvider,
+  ])
 
   const delegationHandler = async (): Promise<void> => {
     await confirmTransaction('delegate')
   }
 
-  const validationError = !!delegationValidationError || !stakePool || !!stakePool.validationError
+  const validationError = !!delegationValidationError || !!error || !stakePool
```

Comment:
why is !stakePool only single negation? :thinking:

Reply:
I assume it should read as `if stake pool is evaluated as false` => `we consider it error situation`. If it was `!!stakePool` we would consider it an error if stake pool was defined. Though maybe we do not need such condition, but I did not want to change logic much in this refactor.